### DISCUSSION TITLE
How to get a pending tx id

### DISCUSSION THREAD

**Discussion by 34r7h** [Python-Cardano/pycardano#14] [Q&A] [answered]
Hi @cffls, i'm able to run the example to create and send a tx via blockfrost but I'd like the txid to be returned from context submit_tx. I naively tried `tx = context.submit_tx(signed_tx.to_cbor())` but returns `None`

I made a pr assuming it's that simple but didn't test. please advise if there's a way already.

**Comment by astrojarred** [comment, marked as answer]
Hey there, that's a great question!

It turns out that on Cardano you can already calculate the Tx ID directly from the signed transaction!
@cffls has conveniently implemented this for us. Once you've signed the tx you can get the resulting transaction ID as a string simply with `tx_id = str(signed_tx.id)`, even before you've submitted it.

Let me know if that ends up working for you.

**Comment by cffls** [comment]
Thanks Jarred @astrojarred ! I also answered the question in @34r7h 's pull request: https://github.com/cffls/pycardano/pull/12#issuecomment-1048127334

**Comment by astrojarred** [comment]
Oh great! Did not see the PR 😄

**Comment by 34r7h** [comment]
thanks, `tx_id = str(signed_tx.id)` works as suggested
